tldw_Server_API/app/core/External_Sources/zotero.py

```
@@ -0,0 +1,584 @@
+        self,
+        raw_item: dict[str, Any],
+        raw_attachments: list[dict[str, Any]],
+        *,
+        collection_key: str | None = None,
+        collection_name: str | None = None,
+        provider_library_id: str | None = None,
+    ) -> NormalizedReferenceItem:
+        data = raw_item.get("data") or {}
+        provider_item_key = str(data.get("key") or raw_item.get("key") or "").strip()
+        if not provider_item_key:
+            raise ValueError("Zotero item is missing a key.")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)